core/scratch_block_comment.js

```diff
@@ -58,17 +58,27 @@ Blockly.ScratchBlockComment = function(block, text, id, x, y, minimized) {
    */
   this.text_ = text;
   /**
-   * The x position of this comment in workspace coordinates.
+   * Whether this comment needs to be auto-positioned (based on provided values
+   * for x and y position).
+   * @type {boolean}
+   * @private
+   */
+  this.needsAutoPositioning_ = !x && x != 0 && !y && y != 0;
+
+  /**
+   * The x position of this comment in workspace coordinates. Default to 0 if
+   * x position is not provided or is not a valid number.
    * @type {number}
    * @private
    */
-  this.x_ = x;
+  this.x_ = typeof x == 'number' && !isNaN(x) ? x : 0;
   /**
-   * The y position of this comment in workspace coordinates.
+   * The y position of this comment in workspace coordinates. Default to 0 if
+   * y position is not provided or is not a valid number.
    * @type {number}
    * @private
    */
-  this.y_ = y;
+  this.y_ = typeof y == 'number' && !isNaN(y) ? y : 0;
```

Comment:
Could you do this check on x/y above and reuse the result of `typeof y == 'number' && !isNaN(y)` to make the line I commented on above more clear?